compiler/plc_diagnostics/src/reporter.rs

```diff
@@ -35,6 +35,7 @@ impl ResolvedLocation {
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct ResolvedDiagnostics {
+    pub code: String,
     pub message: String,
     pub severity: Severity,
     pub main_location: ResolvedLocation,
```

Comment:
again I find "resolved" not optimal. what got resolved? maybe we mean AbsLocation? CanonicalLocation?